tests/test_auth.py

```diff
@@ -227,3 +227,20 @@ def test_warns_for_empty_password(
     assert auth.Resolver(config, auth.CredentialInput()).password == password
 
     assert caplog.messages[0].startswith(warning)
+
+
+def test_log_exception_on_keyring_failure(config, monkeypatch, caplog):
+    class FailKeyring:
+        @staticmethod
+        def get_credential(system, username):
+            import os
+
+            # Let's simulate the error that occurred
+            # in https://github.com/pypa/twine/issues/889
+            os.environ["HOME"]
```

Comment:
The test fails locally if I just run `pytest`, because `HOME` is in `os.environ` (instead of being blocked by tox).

```suggestion
        def get_credential(system, username):
            # Simulate the error from https://github.com/pypa/twine/issues/889
            environ = {}
            environ["HOME"]
```

Reply:
Uhm, interesting, I actually thought that using monkeypatch would be enough to mock this behaviour, and as a matter of fact it works for me locally.

Oh well, I'll just update it as you suggest

Reply:
Ah, I actually thought I was setting up the environment with `monkeypatch.delenv`